[Export] Don't serialize missing args with default value

SherlockNoMad opened the pull request:
Summary: Per https://docs.google.com/document/d/1FzWm-sHYwmRi3x_g036kOxd99KaYquUsA-L5JwOn8ys/edit

I wonder if this would break executorch? @larryliu0820 
I see exir/serialize.py using export's GraphModuleSerializer. 

Test Plan: Existing CIs

Differential Revision: D50519217




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan

zhxchen17 commented on torch/_export/serde/serialize.py:
```diff
@@ -459,7 +459,7 @@ def serialize_sym_op_inputs(self, args) -> List[NamedArgument]:
         return serialized_args
 
     def serialize_inputs(
-        self, target: torch._ops.OpOverload, args, kwargs=None, include_default_values=True
```
@SherlockNoMad I think we should just remove this argument `include_default_value` and the behaviour should always be excluding default argument values out of serialization. I don't think executorch would be an issue here and we can find ways to handle them.

SherlockNoMad replied:
ok.

SherlockNoMad committed:
[Export] Don't serialize missing args with default value (#111715)

Summary:

Per https://docs.google.com/document/d/1FzWm-sHYwmRi3x_g036kOxd99KaYquUsA-L5JwOn8ys/edit

Test Plan: Existing CIs

Differential Revision: D50519217

SherlockNoMad commented:
@pytorchbot merge

larryliu0820 commented:
Sorry for the late reply. I think this won't break ExecuTorch because it's not depending on export serialization. It might change the behavior of ExecuTorch SDK though but I don't think that's a big issue cc @tarun292
